### DISCUSSION TITLE
新機能案

### DISCUSSION THREAD

**Discussion by PrefKarafuto** [PrefKarafuto/ex0ch#14] [Ideas]
新機能の提案はこちらに

**Comment by PrefKarafuto** [comment]
+ NGワードで正規表現を使用可能に  
+ UAフィルター・"＞"による書き込み強調・画像表示プラグイン等を本体に取り込み  
+ NGワード・NGIP等の規制に引っかかった書き込みのログを取得・表示し、かつどこが引っかかったのか管理画面から確認できるようにする  
+ スレ・レス検索機能の改良・掲示板毎にsearch.cgiでの検索可否を設定できるように  
+ 掲示板全体の書き込みを管理画面から検索出来るようにし、一括で削除出来るようにする  
+ 特定スレッド（e.g.管理スレ）にはられた掲示板上レスのリンクを読み込んで、直接リンク先の開示・削除・規制などを可能に

**Reply by PYU224** [reply]
画像とリンク先のプレビューについてなのですが、javascriptを利用すればいけるかもしれません。
JSを切っているTorユーザーには優しくないのですが。

**Reply by PrefKarafuto** [reply]
これ正規表現は通常NGと分けたほうが後々いいでしょうね
どこが正規表現に引っかかったか判りやすくなるし、あとはユーザーの権限で正規表現NGワードを規制できたほうがいいと思います

**Comment by PrefKarafuto** [comment]
+ スマホ対応のレスポンシブデザインに
+ SNSボタンの設置やURLのプレビュー機能、Captcha機能の追加(設定でオンオフができる)
+ スレ立て時にﾜｯﾁｮｲやID無し、強制開示等を設定できる機能
+ コマンドを用いて文字色やフォントを調整する機能
+ レスに対して絵文字で反応できる機能

**Reply by PYU224** [reply]
三男掲示板というのはご存知ですかね？
あれの掲示板スクリプトは0ch+みたいなのでレスポンシブデザインにおいて何か参考になるかもしれません。

**Comment by koushinkyouto** [comment]
・C言語やPython等他のプログラミング言語でもプラグインを作成できるように
・スレやレスを他のSNSに共有できる機能

**Comment by PYU224** [comment]
ここが適切かどうかは分かりませんが・・・
ReCaptcha・hCaptcha認証の設定って需要ありますかね？
需要があれば作ってみようと思いますが。

**Reply by PrefKarafuto** [reply]
あったほうがいいと思います

**Reply by PYU224** [reply]
分かりました。
明日辺りにその機能を搭載してみようと思います。

**Reply by PYU224** [reply]
搭載したものをプルリクエストしました。

**Reply by PrefKarafuto** [reply]
hCapcha適応可否はチェックボックス形式にしてはどうでしょうか

**Reply by PYU224** [reply]
分かりました。
ただ変えないといけないファイルが若干多いので少しお待ちください。

**Reply by PYU224** [reply]
終わりました。
ご確認ください。

**Reply by PrefKarafuto** [reply]
シークレットキーがSETTING.TXTに保存されるのはまずいです

**Reply by PrefKarafuto** [reply]
一応サイトキーと合わせて別ファイル保管にする必要がありますね

**Reply by PrefKarafuto** [reply]
Perl側にもキャプチャモジュールとしてGD::SecurityImageがあったりするので、デフォルトでも使えるキャプチャとして追加するのもありかもしれません。

**Reply by PYU224** [reply]
「シークレットキーがsetting.txtにあるのはまずい」
この件の対策としてシークレットキーを別ファイルにするだけでなく、内容を暗号化して保存する必要があると予測される為に一旦setting.txtに保存したバージョンをアップします。

**Reply by PYU224** [reply]
GDで画像認証はPHPなら雛形がありましたがPerlではあったかどうか定かではありません。

**Comment by PrefKarafuto** [comment]
2ｃｈ時代の旧仕様のオミット・変更も必要ですね

**Reply by PYU224** [reply]
少なくともガラケー向けの機能は切った方が良いかと思います。

**Reply by PrefKarafuto** [reply]
たしかp.cgi、r.cgiとかいうi-mode時代の遺物がありましたね・・・

**Reply by PYU224** [reply]
いじってみてあれらはセキュリティホールになりかねないので削除した方が無難かと思われます。

**Comment by PYU224** [comment]
・書き込み情報等のリアルタイム更新
node.js等が恐らく必要で下手すると全スクリプトが書き直しになるから優先順位は低いです。

**Reply by PYU224** [reply]
普通のJSでいけそうです。
これ見てたら色々応用が効きそうな機能でワクワクする。
https://into-the-program.com/javascript-show-current-date-and-time/

**Reply by PYU224** [reply]
メモと参考用に使えそうなページを載せます。
https://into-the-program.com/javascript-show-current-date-and-time/
https://tech.amefure.com/web-chat-ajax
https://www.nishishi.com/javascript-tips/realtime-clock-setinterval.html
https://1-notes.com/javascript-get-form-values-in-real-time/

以上のページを参考にテストページを作成しました。「適当な名前.html」で動かしてみてください。
入力内容の自動プレビュー機能も搭載できるかもしれません。
```

<p id="RealtimeClockArea2">※ここに時計(2桁固定版)が表示されます。</p>

<p id="output"></p>
<p><textarea id="text_form" value="1" cols="50" rows="5" maxlength="15" placeholder="テキストを入力してください。"></textarea></p>

<script>
function set2fig(num) {
   // 桁数が1桁だったら先頭に0を加えて2桁に調整する
   var ret;
   if( num < 10 ) { ret = "0" + num; }
   else { ret = num; }
   return ret;
}
function showClock2() {
   var nowTime = new Date();
   var nowHour = set2fig( nowTime.getHours() );
   var nowMin  = set2fig( nowTime.getMinutes() );
   var nowSec  = set2fig( nowTime.getSeconds() );
   var msg = "現在時刻は、" + nowHour + ":" + nowMin + ":" + nowSec + " です。";
   document.getElementById("RealtimeClockArea2").innerHTML = msg;
}
setInterval('showClock2()',1000);
</script>


<script>
let text_form = document.querySelector('#text_form');
let output = document.querySelector('#output');

text_form.addEventListener('keyup', function () {
	output.innerHTML = text_form.value;
});
</script>
```

**Comment by PYU224** [comment]
今の今まで存在を忘れていたのですが、てがろぐなるPerlで書かれたスクリプトは参考になるでしょうか。
多機能なので何か参考になるかもしれません。
https://www.nishishi.com/cgi/tegalog/

**Reply by PrefKarafuto** [reply]
これ結構いいですね
tweet埋め込み等参考になります

**Comment by PYU224** [comment]
スレッドタイトルの検索機能
があると嬉しいです。

**Reply by PrefKarafuto** [reply]
あれsearch.cgiからだとスレタイ検索って出来なかったんでしたっけ

**Reply by PYU224** [reply]
確かできなかったような気がします。

**Comment by PrefKarafuto** [comment]
旧ぜろちゃんねるプラス開発陣がやり残したことの完遂
https://ja.osdn.net/projects/zerochplus/ticket

**Reply by PrefKarafuto** [reply]
なお現時点での進捗3/40

**Comment by anime36** [comment]
独自の地域表示は入れることは可能でしょうか？

**Reply by PrefKarafuto** [reply]
IPと地域の紐付けデータベースを導入すれば可能ですね
https://dev.maxmind.com/geoip/geolite2-free-geolocation-data
